core/templates/dev/head/components/state-editor/state-editor-properties-services/state-editor.service.ts

```diff
@@ -53,6 +53,46 @@ export class StateEditorService {
   misconceptionsBySkill: {} = {};
   explorationIsWhitelisted: boolean = false;
   solicitAnswerDetails: boolean = null;
+  stateContentEditorInitialised: boolean = false;
+  stateInteractionEditorInitialised: boolean = false;
+  stateResponsesInitialised: boolean = false;
+  stateHintsEditorInitialised: boolean = false;
+  stateSolutionEditorInitialised: boolean = false;
+  stateEditorDirectiveInitialised: boolean = false;
+
+  updateStateContentEditorInitialised(value: boolean): void {
```

Comment:
This method is only used to mark `stateContentEditorInitialised` to true, right? If so, why do we need the value parameter?

Also, I think this service is for state properties (as per fileoverview) I think we should create a serperate service to handle stateEditorIntialization (which is more of UI centric.)  

I'm not sure about this idea s the name of this file is different than the expected name of the Service. @aks681, Will it be fine to leave these functions in this service?

Reply:
I've added the function here because StateEditorService is used in all the state editor components.

Reply:
I feel it might involve some bookkeeping if we decide to move it to a new service (ie. ensuring all isInitialised variables are set to false when it is not required), so leaving that as is for now.